crates/fuzzing/src/oracles/memory.rs

```diff
@@ -0,0 +1,396 @@
+//! Oracles related to memory.
+
+use crate::generators::{HeapImage, MemoryAccesses};
+use wasmtime::*;
+
+/// Oracle to perform the described memory accesses and check that they are all
+/// in- or out-of-bounds as expected
+pub fn check_memory_accesses(input: MemoryAccesses) {
+    crate::init_fuzzing();
+    log::info!("Testing memory accesses: {input:#x?}");
+
+    let offset = input.offset;
+    let growth = input.growth;
+    let wasm = build_wasm(&input.image, offset);
+    crate::oracles::log_wasm(&wasm);
+    let offset = u64::from(offset);
+
+    let mut config = input.config.to_wasmtime();
+
+    // Force-enable the memory64 proposal if the heap image wants it.
+    if input.image.memory64 {
+        config.wasm_memory64(true);
+    }
+
+    let engine = Engine::new(&config).unwrap();
+    let module = match Module::new(&engine, &wasm) {
+        Ok(m) => m,
+        Err(e) => {
+            let e = format!("{e:?}");
+            log::info!("Failed to create `Module`: {e}");
+            assert!(
+                e.contains("bytes which exceeds the configured maximum of")
+                    || e.contains("exceeds the limit of")
+            );
+            return;
+        }
+    };
+
+    let limits = super::StoreLimits::new();
+    let mut store = Store::new(&engine, limits);
```

Comment:
This can use `input.config.to_store()` which should also handle things like epochs/fuel configuration I believe.

Reply:
Unfortunately, we can't use that since we are setting extra config settings based on the arbitrary heap image, but we don't have access to the config/engine if we use that method.